Add Swift wrapper header

adetaylor opened the pull request:
#### 065d53aea21b7758aa8ad917aa3a291d271b3cab
<pre>
Add Swift wrapper header
<a href="https://bugs.webkit.org/show_bug.cgi?id=302931">https://bugs.webkit.org/show_bug.cgi?id=302931</a>
<a href="https://rdar.apple.com/165200310">rdar://165200310</a>

Reviewed by Geoffrey Garen and Elliott Williams.

In future various parts of WebKit&apos;s C++ will want to include the generated
definitions/declarations of Swift types and functions. This is already the case
with the Swift demo feature and will become more important with upcoming
CoreIPC work. To do this, C++ needs to refer to a header file emitted by Swift.
That header file has a few quirks which make it awkward to include:

* It generates warnings which WebKit treats as errors.
* It does not include pre-requisite headers, e.g. for the types used as
  parameters to Swift functions.
* The WebKit build process may generate the header with two different names
  depending on SDK version.

This commit introduces a &apos;wrapper&apos; header which resolves all these problems in
one place, so that any other code wishing to refer to Swift definitions and
declarations can simply #include &quot;WebKit-Swift.h&quot;. At the moment this is not a
dramatic simplification, but it will become so as Swift usage expands.

Canonical link: <a href="https://commits.webkit.org/303477@main">https://commits.webkit.org/303477@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4c0a3dfc55ae815ce1af63a3ad50a65fbe91e667

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/132545 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5040 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43605 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140064 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/84545 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e09faff8-e864-4877-89cb-ca45a48a86d2) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/5199 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4799 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101332 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/68621 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/2dcaa083-8321-43e2-a3c6-623362885154) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/135491 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3663 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/118733 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82128 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/fc5db157-3988-41bd-a202-abfd6394ceaa) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/3549 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1325 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/83299 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/112452 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142718 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/4710 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37437 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/109708 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/4792 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4071 "Found 1 new test failure: imported/w3c/web-platform-tests/mediacapture-streams/overconstrained_error.https.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109890 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27846 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/3582 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115005 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/58133 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/4764 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33355 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/4599 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68215 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/4855 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4721 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Source/WebKit/Shared/WebKit-Swift.h:
```diff
@@ -25,22 +25,23 @@
+
+// rdar://165192318
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warc-bridge-casts-disallowed-in-nonarc"
```
Slightly nicer to use `IGNORE_CLANG_WARNINGS_BEGIN("arc-bridge-casts-disallowed-in-nonarc")` / `IGNORE_CLANG_WARNINGS_END`.

adetaylor replied:
Good point, done

adetaylor committed:
Add Swift wrapper header
https://bugs.webkit.org/show_bug.cgi?id=302931
rdar://165200310

Reviewed by Geoffrey Garen and Elliott Williams.

In future various parts of WebKit's C++ will want to include the generated
definitions/declarations of Swift types and functions. This is already the case
with the Swift demo feature and will become more important with upcoming
CoreIPC work. To do this, C++ needs to refer to a header file emitted by Swift.
That header file has a few quirks which make it awkward to include:

* It generates warnings which WebKit treats as errors.
* It does not include pre-requisite headers, e.g. for the types used as
  parameters to Swift functions.
* The WebKit build process may generate the header with two different names
  depending on SDK version.

This commit introduces a 'wrapper' header which resolves all these problems in
one place, so that any other code wishing to refer to Swift definitions and
declarations can simply #include "WebKit-Swift.h". At the moment this is not a
dramatic simplification, but it will become so as Swift usage expands.

Canonical link: https://commits.webkit.org/303477@main
